disnake/interactions/base.py

```diff
@@ -679,23 +734,30 @@ async def defer(self, *, ephemeral: bool = False, with_message: bool = False) ->
         if self._responded:
             raise InteractionResponded(self._parent)
 
-        defer_type: int = 0
-        data: Optional[Dict[str, Any]] = None
+        defer_type: Optional[int] = None
+        data: Dict[str, Any] = {}
         parent = self._parent
-        if (
-            parent.type in (InteractionType.application_command, InteractionType.modal_submit)
-            or with_message
-        ):
+        if with_message is MISSING:
+            if ephemeral is not MISSING:
+                with_message = True
+
+        if parent.type in (InteractionType.application_command, InteractionType.modal_submit):
             defer_type = InteractionResponseType.deferred_channel_message.value
-            if ephemeral:
-                data = {"flags": 64}
         elif parent.type is InteractionType.component:
-            defer_type = InteractionResponseType.deferred_message_update.value
+            if with_message:
+                defer_type = InteractionResponseType.deferred_channel_message.value
+            else:
+                defer_type = InteractionResponseType.deferred_message_update.value
+
+        if defer_type == InteractionResponseType.deferred_channel_message.value:
+            data["flags"] = 0
```

Comment:
Nit: Comment that this is because we're not sending a message, perhaps? 

"Reset flags as we're not actually sending a message and these flags are thus purposeless"

Reply:
https://github.com/DisnakeDev/disnake/pull/474/commits/1517569c6b3b48cbe5a37df07a10b8a55e3d530d